lib/diplomat/key.ex

```diff
@@ -175,18 +154,37 @@ defmodule Diplomat.Key do
     get([key])
   end
 
-  def urlsafe(%__MODULE__{} = key) do
-    key
-    |> proto
-    |> PbKey.encode
-    |> Base.url_encode64(padding: false)
+  @spec path_to_proto([key_pair], [PbPathElement.t]) :: [PbPathElement.t]
+  defp path_to_proto([], acc), do: acc
+  defp path_to_proto([[kind, id]|tail], acc) when is_nil(id) do
+    path_to_proto(tail, [PbPathElement.new(kind: kind, id_type: nil)|acc])
+  end
+  defp path_to_proto([[kind, id]|tail], acc) when is_integer(id) do
+    path_to_proto(tail, [PbPathElement.new(kind: kind, id_type: {:id, id})|acc])
+  end
+  defp path_to_proto([[kind, name]|tail], acc) do
+    path_to_proto(tail, [PbPathElement.new(kind: kind, id_type: {:name, name})|acc])
   end
```

Comment:
Sorry for the churn here 😟. I've grouped private functions at the bottom of the file. If you would prefer I didn't it's not a problem I can roll it back.